processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -159,8 +159,18 @@ public GroupByQueryQueryToolChest(
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
       final List<AggregatorFactory> aggs = Lists.newArrayList();
+
       for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
```

Comment:
@dclim I think it's OK to throw an exception if your outer query refers to a field you didn't actually generate in your inner query. We don't usually throw errors for nonexistent columns in Druid, but that's mostly because we want to make it easy for people to evolve their schemas. For something like this the existence of the inner column is totally under the queryer's control.